diesel/src/sqlite/connection/mod.rs

```diff
@@ -351,7 +357,15 @@ impl SqliteConnection {
             &source,
             &Sqlite,
             &[],
-            |sql, is_cached| Statement::prepare(raw_connection, sql, is_cached),
+            |sql, counter| {
```

Comment:
I would prefer forwarding the cache information explicitly here instead of relaying on the workaround using the optional counter. It should be fine to just have it as third function argument.